### ISSUE TITLE
Consistent Learning Rate Parameters Requirements

### ISSUE THREAD

**Issue by Deepblue129** [pytorch/pytorch#2085]
**Background:**
I am implementing a CLI to use a Pytorch model. If the user sets the `lr` to `None` it defaults to Pytorch's default `lr` per different optimizers. This works well except for `torch.optim.SGD` as it requires a `lr` parameter instead of defaulting to 1. This behavior is not consistent with every other `torch.optim.*` and forces me to handle a one-off case. 

**Issue:**
`torch.optim.SGD` unlike every other `torch.optim.*` has `lr` as a non-optional parameter.  For consistency, can Pytorch always require `lr` or never require it. `torch.optim.SGD` seems like a one-off case.

**Comment by soumith** [comment]
it makes sense for something like `Adam` to have a default learning rate, but it doesn't make sense at all for `SGD`. The range of learning rates usually explored for Adam (for example) are very small and some discrete choices, within very close values of each other. It's not the case for SGD.
I dont think it makes sense for us to make the `lr` be consistently required or not-required (some optimizers like YellowFin dont even have a learning rate).

**Comment by apaszke** [comment]
As @soumith explained it's intentional
